src/wwwroot/controllers/SucceedRegistrationCtrl.js

```diff
@@ -6,16 +6,49 @@
       .controller('SucceedRegistrationCtrl', SucceedRegistrationCtrl);
   
       SucceedRegistrationCtrl.$inject = [
-      'utils'
+      'auth',
+      '$translate',
+      '$location',
+      'vcRecaptchaService'
     ];
   
-    function SucceedRegistrationCtrl(utils) {
+    function SucceedRegistrationCtrl(auth, $translate, $location, vcRecaptchaService) {
         var vm = this;
-        vm.getCurrentLanguage = getCurrentLanguage;
+        var queryParams = $location.search();
+        var email = queryParams['email'];
+        if (!email) {
+            $location.path('/');
+        }
+        vm.resendEmail = resendEmail;
+        vm.resendFinished = false;
+        vm.setResponse = setResponse;
+        vm.setWidgetId = setWidgetId;
+        vm.captchaExpiration = captchaExpiration;
+
+        function setWidgetId (widgetId) {
+            vm.widgetId = widgetId;
+        };
+        
+        function captchaExpiration () {
+            vcRecaptchaService.reload(vm.widgetId);
+            vm.response = null;
+        };
+
+        function resendEmail() {
+            vcRecaptchaService.reload(vm.widgetId);
+            vcRecaptchaService.execute(vm.widgetId);
+        }
 
-        function getCurrentLanguage() {
-            return utils.getPreferredLanguage();
+        function setResponse (response) {
+            auth.forgotPassword(email, $translate.use(), response)
+            .then(function (result) {
+                 vm.resendFinished = true;
```

Comment:
If I understand it well, this line hides the "resend button".

In that case, I think that we should apply it just before call backend, for example:

```js
function setResponse (response) {
    vm.resendFinished = true;
    auth
        .forgotPassword(email, $translate.use(), response)
        .catch(function() {
            captchaExpiration();
            vm.resendFinished = false;
        });
}
```

Another option could be show a spinner, but I think that it is not necessary, the user does not need to know exactly when the email have been sent, at least not at millisecond.

Maybe we could also rename `resendFinished` as `resent`